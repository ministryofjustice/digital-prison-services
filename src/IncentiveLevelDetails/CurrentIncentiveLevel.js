import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import GridRow from '@govuk-react/grid-row'
import GridCol from '@govuk-react/grid-col'
import Button from '@govuk-react/button'
import { BLUE } from 'govuk-colours'
import CurrentIncentiveLevelArea from './IncentiveLevelDetails.styles'

const CurrentIncentiveLevel = ({ level, days, nextReviewDate, userCanMaintainIep, history }) => (
  <CurrentIncentiveLevelArea className="current-iep">
    <GridRow>
      <GridCol setWidth="one-quarter">
        <strong className="label">Current incentive level</strong>
        <p>{level}</p>
      </GridCol>
      <GridCol setWidth="one-quarter">
        <strong className="label">Time since review</strong>
        <p>{days}</p>
      </GridCol>
      <GridCol setWidth="one-quarter">
        <strong className="label">Date of next review</strong>
        <p>{nextReviewDate}</p>
      </GridCol>
      {userCanMaintainIep && (
        <GridCol setWidth="one-quarter">
          <div>
            <Button
              buttonColour={BLUE}
              onClick={() => history.push(`incentive-level-details/change-incentive-level`)}
              data-qa="change-incentive-level"
            >
              Change incentive level
            </Button>
          </div>
        </GridCol>
      )}
    </GridRow>
  </CurrentIncentiveLevelArea>
)

CurrentIncentiveLevel.propTypes = {
  level: PropTypes.string,
  days: PropTypes.string,
  nextReviewDate: PropTypes.string,
  history: PropTypes.shape({
    replace: PropTypes.func.isRequired,
    push: PropTypes.func.isRequired,
  }).isRequired,
  userCanMaintainIep: PropTypes.bool.isRequired,
}

CurrentIncentiveLevel.defaultProps = {
  level: '',
  days: '',
  nextReviewDate: '',
}

const mapStateToProps = state => ({
  level: state.iepHistory.currentIepLevel,
  days: state.iepHistory.daysOnIepLevel,
  nextReviewDate: state.iepHistory.nextReviewDate,
  userCanMaintainIep: state.iepHistory.userCanMaintainIep,
})

export default connect(mapStateToProps)(CurrentIncentiveLevel)
