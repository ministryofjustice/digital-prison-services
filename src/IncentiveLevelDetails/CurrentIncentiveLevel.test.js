import React from 'react'
import testRenderer from 'react-test-renderer'
import CurrentIncentiveLevel from './CurrentIncentiveLevel'

const initialState = {
  iepHistory: {
    currentIepLevel: 'Standard',
    daysOnIepLevel: '625',
    currentIepDateTime: '2017-08-15T16:04:35',
    nextReviewDate: '15/08/2018',
    establishments: [{ agencyId: 'LEI', description: 'Leeds' }],
    levels: ['Standard'],
    userCanMaintainIep: false,
    results: [
      {
        bookingId: -1,
        iepDate: '2017-08-13',
        iepTime: '2017-08-13T16:04:35',
        formattedTime: '13/08/2017 - 16:04',
        iepEstablishment: 'Leeds',
        iepStaffMember: 'Staff Member',
        agencyId: 'LEI',
        iepLevel: 'Standard',
        userId: 'ITAG_USER',
      },
    ],
  },
}

describe('Current incentive level', () => {
  const store = {}
  const history = {}
  beforeEach(() => {
    store.getState = jest.fn()
    store.subscribe = jest.fn()
    store.dispatch = jest.fn()
    history.push = jest.fn()
    history.replace = jest.fn()
  })

  describe('should render the current Incentive Level correctly', () => {
    it("when user can't maintain Incentive Level", () => {
      store.getState.mockReturnValue(initialState)
      const wrapper = testRenderer.create(<CurrentIncentiveLevel history={history} store={store} />).toJSON()

      expect(wrapper).toMatchSnapshot()
    })

    it('when user can maintain Incentive Level', () => {
      store.getState.mockReturnValue({
        ...initialState,
        iepHistory: { ...initialState.iepHistory, userCanMaintainIep: true },
      })
      const wrapper = testRenderer.create(<CurrentIncentiveLevel history={history} store={store} />).toJSON()

      expect(wrapper).toMatchSnapshot()
    })
  })
})
