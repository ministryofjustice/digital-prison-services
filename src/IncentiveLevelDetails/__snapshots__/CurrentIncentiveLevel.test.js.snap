// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current incentive level should render the current Incentive Level correctly when user can maintain Incentive Level 1`] = `
<div
  className="sc-ifAKCX kQggjh current-iep"
>
  <div
    className="sc-bdVaJa iAgRsv"
  >
    <div
      className="sc-bwzfXH mIHzr"
    >
      <strong
        className="label"
      >
        Current incentive level
      </strong>
      <p>
        Standard
      </p>
    </div>
    <div
      className="sc-bwzfXH mIHzr"
    >
      <strong
        className="label"
      >
        Time since review
      </strong>
      <p>
        625
      </p>
    </div>
    <div
      className="sc-bwzfXH mIHzr"
    >
      <strong
        className="label"
      >
        Date of next review
      </strong>
      <p>
        15/08/2018
      </p>
    </div>
    <div
      className="sc-bwzfXH mIHzr"
    >
      <div>
        <button
          className="sc-htpNat iXzDIq"
          data-qa="change-incentive-level"
          disabled={false}
          onClick={[Function]}
        >
          Change incentive level
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Current incentive level should render the current Incentive Level correctly when user can't maintain Incentive Level 1`] = `
<div
  className="sc-ifAKCX kQggjh current-iep"
>
  <div
    className="sc-bdVaJa iAgRsv"
  >
    <div
      className="sc-bwzfXH mIHzr"
    >
      <strong
        className="label"
      >
        Current incentive level
      </strong>
      <p>
        Standard
      </p>
    </div>
    <div
      className="sc-bwzfXH mIHzr"
    >
      <strong
        className="label"
      >
        Time since review
      </strong>
      <p>
        625
      </p>
    </div>
    <div
      className="sc-bwzfXH mIHzr"
    >
      <strong
        className="label"
      >
        Date of next review
      </strong>
      <p>
        15/08/2018
      </p>
    </div>
  </div>
</div>
`;
